cpp/src/io/utilities/hostdevice_vector.hpp

```diff
@@ -96,6 +96,14 @@ class hostdevice_vector {
   [[nodiscard]] size_t size_bytes() const noexcept { return sizeof(T) * size(); }
   [[nodiscard]] bool empty() const noexcept { return size() == 0; }
 
+  void resize(size_t new_size, rmm::cuda_stream_view stream)
+  {
+    CUDF_EXPECTS(new_size <= capacity(),
+                 "hostdevice_vector resize must fit within existing capacity");
+    stream.synchronize();
```

Comment:
Why?

Reply:
I wanted to avoid the complexity of having to actually reallocate/copy existing elements since the use case here is very narrow.

Reply:
Also, as @vuule pointed out down below, the usage for this is unnecessary.  An earlier iteration of the code had the vector initialized early on and then resized downwards later.  But in the current iteration, it is all done in one place so this change is not necessary. I'll remove it.